libres/tests/enkf/test_enkf_fs.cpp

```diff
@@ -0,0 +1,55 @@
+#include <filesystem>
+
+#include "catch2/catch.hpp"
+
+#include <ert/enkf/enkf_fs.hpp>
+#include <ert/enkf/enkf_obs.hpp>
+
+#include "../tmpdir.hpp"
+
+void enkf_fs_fwrite_misfit(enkf_fs_type *fs);
+enkf_fs_type *enkf_fs_alloc_empty(const char *mount_point);
+extern "C" void enkf_fs_umount(enkf_fs_type *fs);
+extern "C" void
+misfit_ensemble_initialize(misfit_ensemble_type *misfit_ensemble,
+                           const ensemble_config_type *ensemble_config,
+                           const enkf_obs_type *enkf_obs, enkf_fs_type *fs,
+                           int ens_size, int history_length, bool force_init);
+void enkf_fs_init_path_fmt(enkf_fs_type *fs);
+void enkf_fs_set_read_only(enkf_fs_type *fs, bool read_only);
+
+TEST_CASE("enkf_fs_fwrite_misfit", "[enkf]") {
+    GIVEN("An instance of enkf_fs") {
+        TmpDir tmpdir;
```

Comment:
Minor style stuff: I have a macro which does this but doesn't require you to give `TmpDir` a name.
```suggestion
        WITH_TMPDIR;
```